packages/block-library/src/site-logo/edit.js

```diff
@@ -0,0 +1,283 @@
+	} );
+
+	const classes = classnames( 'custom-logo-link', {
+		'is-transient': isBlobURL( url ),
+	} );
+
+	const img = (
+		<a href="#home" className={ classes } rel="home">
+			<img className="custom-logo" src={ url } alt={ alt } />
+		</a>
+	);
+
+	const label = __( 'Site Logo' );
+	const logoImage = (
+		<ImageSize src={ url }>
+			{ ( sizes ) => {
+				const {
+					imageWidthWithinContainer,
+					imageWidth,
+					imageHeight,
+				} = sizes;
+
+				const currentWidth = width || imageWidthWithinContainer;
+				const ratio = imageWidth / imageHeight;
+				const currentHeight = currentWidth / ratio;
+
+				const maxWidthProp = {};
+				if ( maxWidth ) {
+					maxWidthProp.maxWidth = Math.min( imageWidth, maxWidth );
+				}
+
+				const canResize = ! isEditing && isLargeViewport && ( ! width || width <= maxWidth );
+
+				const wrapperProps = {};
+				if ( align ) {
+					wrapperProps.className = `align${ align }`;
+				}
+
+				if ( ! canResize ) {
+					wrapperProps.style = { width: Math.min( currentWidth, maxWidth ) };
+				}
+
+				const boxSize = {
+					width: Math.min( currentWidth, maxWidth ),
+					height: ( maxWidth && currentWidth <= maxWidth ) ? currentHeight : maxWidth / ratio,
+				};
+
+				return (
+					<>
+						{ getInspectorControls( imageWidth, width, maxWidth, canResize ) }
+						<div { ...wrapperProps }>
+							{ canResize ? (
+								<ResizableBox
+									showHandle={ isSelected }
+									size={ boxSize }
+									lockAspectRatio={ true }
+									minWidth={ 10 }
+									{ ...maxWidthProp }
+									enable={ {
+										top: false,
+										bottom: true,
+										...getHandleStates( align, isRTL ),
+									} }
+
+									onResizeStop={ ( event, direction, elt, delta ) => {
+										setAttributes( {
+											width: parseInt( currentWidth + delta.width, 10 ),
+										} );
+									} }
+									style={ { display: 'inline-block' } }
+								>
+									{ img }
+								</ResizableBox>
+							) : img }
+						</div>
+					</>
+				);
+			} }
+		</ImageSize>
+	);
+	const editComponent = (
+		<MediaPlaceholder
+			icon={ <BlockIcon icon={ icon } /> }
+			labels={ {
+				title: label,
+				instructions: __( 'Upload an image, or pick one from your media library, to be your site logo' ),
+			} }
+			onSelect={ onSelectLogo }
+			accept={ ACCEPT_MEDIA_STRING }
+			allowedTypes={ ALLOWED_MEDIA_TYPES }
+			mediaPreview={ !! url && img }
+			notices={ error && (
+				<Notice status="error" isDismissible={ false }>{ error }</Notice>
+			) }
+			onCancel={ !! url && ( () => setIsEditing( false ) ) }
+			onError={ onUploadError }
+		>
+			{ !! url && (
+				<IconButton isLarge icon="delete" onClick={ deleteLogo }>
+					{ __( 'Delete Site Logo' ) }
+				</IconButton>
+			) }
+			{ children }
+		</MediaPlaceholder>
+	);
+
+	return (
+		<div className={ className }>
+			{ controls }
+			{ ! url || isEditing ? editComponent : logoImage }
+		</div>
+	);
+}
```

Comment:
Feels like there's a few things in common between this block and the image  block. Not for this PR but it would be good to find a good way to share behavior.

Reply:
Yes, I have extracted this to another component, but I think it would be good to share this code somehow...